knack/commands.py

```diff
@@ -124,7 +124,12 @@ def update_argument(self, param_name, argtype):
         # that coincides with the default
         if isinstance(arg_default, str):
             arg_default = DefaultStr(arg_default)
-        elif isinstance(arg_default, int):
+        elif isinstance(arg_default, int) and not isinstance(arg_default, bool):
+            # adjust here is because:
+            # 1) bool is subclass of int so isinstance(True, int) cannot distinguish between int and bool
+            # 2) bool is not extendable according to
+            #   https://stackoverflow.com/questions/2172189/why-i-cant-extend-bool-in-python,
+            #   so bool's is_default is ignored for now
```

Comment:
Code comment should usually not focus on adjustment, but the status quo.

We can get the short link for a Stackoverflow post via the "share" button:

![image](https://github.com/microsoft/knack/assets/4003950/eaec2dfc-ef23-4d19-9650-45ae38d5546c)

```suggestion
            # bool's is_default is ignored for now, because
            # 1. bool is a subclass of int, so isinstance(True, int) cannot distinguish between int and bool
            # 2. bool is not extendable according to https://stackoverflow.com/a/2172204/2199657
```